src/arithmetic/arithmetic_expression.c

```diff
@@ -1320,6 +1320,37 @@ SIValue AR_TIMESTAMP(SIValue *argv, int argc) {
 	return SI_LongVal(TemporalValue_NewTimestamp());
 }
 
+//==============================================================================
+//=== Scalar functions =========================================================
+//==============================================================================
+
+SIValue AR_RANDOMUUID(SIValue *argv, int argc) {
```

Comment:
Following a bit more performant version
`SIValue AR_RANDOMUUID(SIValue *argv, int argc) {
	/* Implementation is based on https://www.cryptosys.net/pki/uuid-rfc4122.html */

	// Generate 16 random bytes.
	unsigned char b[16];
	int nbytes = sizeof(int);
	for (int i = 0; i < 16; i += nbytes) {
		int *v = (int*)(b + i);
		*v = rand();
	}

	// Set the four most significant bits of the 7th byte to 0100'B, so the high nibble is "4".
	b[6] = (b[6] & 0b00001111) | 0b01000000;

	// Set the two most significant bits of the 9th byte to 10'B, so the high nibble will be one of "8", "9", "A", or "B" (see Note 1).
	b[8] = (b[8] & 0b00111111) | 0b10000000;

	char *uuid = rm_malloc(37 * sizeof(char));
	sprintf(uuid, "%08x-%04x-%04x-%04x-%04x%08x", *((uint32_t*)b),
		*((uint16_t*)(b + 4)),
		*((uint16_t*)(b + 6)),
		*((uint16_t*)(b + 8)),
		*((uint16_t*)(b + 10)),
		*((uint32_t*)b + 12));

	uuid[36] = '\0';
	return SI_TransferStringVal(uuid);
}`

Reply:
>
```
for (int i = 0; i < 16; i += nbytes) {
	int *v = (int*)(b + i);
	*v = rand();
}
```

I didn't go with this, because `rand()` function is guaranteed to only be in the range from 0 to 32767

Reply:
BTW, I didn't find the `srand(..)` call, but `rand()` has already been used for `rand` cypher function

Reply:
There is an issue with endianness in your implementation. Since we change bits which will later be treated as longer words, the different endianness may change the order. I'll provide another implementation